CRM/MembersOnlyEvent/Test/Fabricator/Event.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+use CRM_Event_DAO_Event as Event;
+
+class CRM_MembersOnlyEvent_Test_Fabricator_Event {
+
+  /**
+   * Fabricates a membership type.
+   *
+   * @param array $params
+   *
+   */
+  public static function fabricate($params = []) {
+    $params = array_merge(static::getDefaultParams(), $params);
+
+    $event = new Event();
+    foreach ($params as $property => $value) {
+      $event->$property = $value;
+    }
+
+    return $event->save();
+  }
+
+  private static function getDefaultParams() {
+    return [
+      "title" => "Test Event",
+      "event_title" => "Test Event",
+      "event_description" => "",
+      "event_type_id" => "2",
```

Comment:
is this event type always guaranteed to be there ?

Reply:
@omarabuhussein  Civicrm default installation ships with  pre-set event types so event_type_id 2  should be existed when executing the test.  However, if the test fails in the future because of event_type_id does not exist, as Civicrm remove default value for the core, we could add event type fabricator to add event type before fabricating an event.